api/envoy/api/v2/cds.proto

```diff
@@ -232,6 +233,13 @@ message Cluster {
   // specific options.
   map<string, google.protobuf.Struct> extension_protocol_options = 35;
 
+  // [#not-implemented-hide:]
+  // The extension_protocol_options field is used to provide extension-specific protocol options
+  // for upstream connections. The key should match the extension filter name, such as
+  // "envoy.filters.network.thrift_proxy". See the extension's documentation for details on
+  // specific options.
+  map<string, google.protobuf.Any> typed_extension_protocol_options = 36;
```

Comment:
Should we `oneof` this with `extension_protocol_options`? You can do this without breaking wire compat.

Reply:
We'll need to have some standard code to check for mutually exclusive use of these then, sure.